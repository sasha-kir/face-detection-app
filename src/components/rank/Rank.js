import React from 'react';

const Rank = ({name, entries}) => {
	return (
		<div className="center-content" style={{justifySelf: "start"}}>
			<div id="name" className="center-content white f3">
				{`${name}, your current entry count is`} &nbsp;
			</div>
			<div id="entries" className="white f1">
				{entries}
			</div>
		</div>
	);
}

export default Rank;