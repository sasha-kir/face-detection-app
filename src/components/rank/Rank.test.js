import React from 'react';
import { mount, shallow } from 'enzyme';
import Rank from './Rank';

describe('Rank', () => {
  it('renders correctly', () => {
    const component = shallow(<Rank />);
    expect(component).toMatchSnapshot();
  });

  it('displays name', () => {
    const component = mount(<Rank name="John" />);
    const greeting = component.find('div#name');
    expect(greeting.text().includes("John")).toBeTruthy();
    component.unmount();
  });

  it('displays number of entries', () => {
    const component = mount(<Rank entries="5" />);
    const entries = component.find('div#entries');
    expect(entries.text().includes("5")).toBeTruthy();
    component.unmount();
  });
});