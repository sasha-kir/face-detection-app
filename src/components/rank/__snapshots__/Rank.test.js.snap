// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rank renders correctly 1`] = `
<div
  className="center-content"
  style={
    Object {
      "justifySelf": "start",
    }
  }
>
  <div
    className="center-content white f3"
    id="name"
  >
    undefined, your current entry count is
     Â 
  </div>
  <div
    className="white f1"
    id="entries"
  />
</div>
`;
